nisystemlink/clients/core/_http_configuration_manager.py

```diff
@@ -78,12 +79,22 @@ def _fallback(cls) -> Optional[core.HttpConfiguration]:
         """
         if cls._configs is None:
             cls._configs = cls._read_configurations()
+
+        if cls._juypter_config is None:
```

Comment:
Can we combine these blocks so we only try to load the Jupyter config once even if it fails?

```suggestion
```

Reply:
@spanglerco, is the concern that when running outside JupyterHub, the JupyterHttpConfig will be attempted every time a client creates and obtains the HttpConfiguration?

Reply:
Yes

Reply:
Refactored into a `cls._virtual_config` to load once like the file-based configs.